wrappers/rust/icicle-core/src/symbol/mod.rs

```
@@ -0,0 +1,346 @@
+              Err(eIcicleError::AllocationFailed)
+            } else {
+              Ok(handle)
+            }
+          }
+        }
+
+        fn mul_handles(op_a: HandleCPP, op_b: HandleCPP) -> Result<HandleCPP, eIcicleError> {
+          unsafe {
+            let mut handle = std::ptr::null();
+            let ffi_status = ffi_multiply_symbols(op_a, op_b, &mut handle);
+            if ffi_status != eIcicleError::Success {
+              Err(ffi_status)
+            } else if handle.is_null() {
+              Err(eIcicleError::AllocationFailed)
+            } else {
+              Ok(handle)
+            }
+          }
+        }
+
+        fn add_field(self, other: $field) -> Result<Self, eIcicleError> {
+          let other_symbol = Self::from_constant(other)?;
+          let res_handle = Self::add_handles(self.m_handle, other_symbol.m_handle)?;
+          Ok(Self { m_handle: res_handle })
+        }
+
+        fn sub_field(self, other: $field) -> Result<Self, eIcicleError> {
+          let other_symbol = Self::from_constant(other)?;
+          let res_handle = Self::sub_handles(self.m_handle, other_symbol.m_handle)?;
+          Ok(Self { m_handle: res_handle })
+        }
+
+        fn mul_field(self, other: $field) -> Result<Self, eIcicleError> {
+          let other_symbol = Self::from_constant(other)?;
+          let res_handle = Self::mul_handles(self.m_handle, other_symbol.m_handle)?;
+          Ok(Self { m_handle: res_handle })
+        }
+      }
+
+      impl Handle for FieldSymbol {
+        fn handle(&self) -> HandleCPP { self.m_handle }
+      }
+
+      impl DeletableHandle for FieldSymbol {
+        fn delete_handle(handle: HandleCPP) {
+          unsafe {
+            if !handle.is_null()
+            {
+              unsafe {
+                let ffi_status = ffi_delete_symbol(handle);
+                if ffi_status != eIcicleError::Success {
+                  panic!("Couldn't delete symbol, due to {:?}", ffi_status);
```

why panic? you can return the error instead

---

Function removed